apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/cql/builder/QueryBuilderImpl.java

```diff
@@ -1346,6 +1368,60 @@ private void appendConditions(
     }
   }
 
+  private void appendConditions(
+      Expression<BuiltCondition> whereExpression, String initialPrefix, StringBuilder builder) {
+    String prefix = initialPrefix;
+    if (initialPrefix.contains("IF") && ifExists) {
+      builder.append(prefix).append("EXISTS");
+      prefix = " AND ";
+    }
+    builder.append(initialPrefix); // must have where
+    builder.append(addExpressionCql(whereExpression));
```

Comment:
I think that you should pass `builder` and append directly onto it, instead of constructing new one, returning -- doing so would avoid allocation of a StringBuilder, copying etc.